lib/person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def can_use_services?
+    is_of_age? || @parent_permission ? true : false
+  end
```

Comment:
- [ ] You do not have a method called `is_of_age?`. What you have is `of_age?`. Kindly fix that.

- [ ] The instance variable `@parent_permission` is either `true` or `false`, so we don't need to check for that.

  ```
  def can_use_services?
    of_age? || @parent_permission
  end
  ```

Reply:
Thank you so much @TedLivist!

Reply:
I don't really understand this **The instance variable @parent_permission is either true or false, so we don't need to check for that.**. I think we have to check that whether the parent permission is okay or not to provide the services, isn't it?

Reply:
Hi @melashu when initializing a person the value of `@parent_permission` is always either `true` or `false` in itself, so it is kind of repetitive to still do this check here.

Just leave it as `@parent_permission` without `? true : false`